src/main/java/com/glencoesoftware/bioformats2raw/HCSIndex.java

```diff
@@ -29,7 +29,7 @@ public HCSIndex(IMetadata meta, int series) {
     for (int p=0; p<meta.getPlateCount(); p++) {
       for (int w=0; w<meta.getWellCount(p); w++) {
         for (int ws=0; ws<meta.getWellSampleCount(p, w); ws++) {
-          if (meta.getWellSampleIndex(p, w, ws).getValue() == series) {
+          if (meta.getWellSampleImageRef(p, w, ws) == meta.getImageID(series)) {
```

Comment:
This comparison should use `.equals` instead of `==`.

Reply:
Should be fixed in 029df918a09fa61a9db9d9c161926e6b389babc6